### PR TITLE
first pass at @link documentation

### PR DESCRIPTION
**Pull request by clenfest**

### CODE REVIEW

File: docs/source/federation-spec.md [hunk 1]
```diff
@@ -309,3 +309,38 @@ extend type User @key(fields: "email") {
...
+
+The `@link` directive provides a mechanism to link to external schemas, specifying the URL, version, and directives. The presence of a `@link` directive also indicates that the schema is a core schema. 
+
+```graphql
+    schema
+      @link(url: "https://specs.apollo.dev/link/v1.0")
+      @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key", "@requires", "@provides", "@external", { name: "@tag", as "@mytag" }, "@extends", "@shareable", "@inaccessible", "@override"])
+```
+ 
+When present, the `import` argument will assign non-prefixed (local) names to elements from a foreign schema. In the example above, `@tag` is renamed to `@mytag` while the other directive names will be assigned as they are specified in the federation schema. The ability to rename directives may be necessary if there is a naming collision in two different foreign schemas.
```
Comment (cpeacock) [inline docs/source/federation-spec.md:346]: Do you think talking about redefining directives in here would be useful or purposely leaving it out is the right way to go?  I do not have a strong opinion, but just musing.

File: docs/source/federation-spec.md [hunk 2]
```diff
@@ -309,3 +309,38 @@ extend type User @key(fields: "email") {
...
+
+directive @link(
+  url: String, 
+  as: String, 
+  for: link__Purpose, 
+  import: [link__Import]
+) repeatable on SCHEMA
+```
+
+The `@link` directive provides a mechanism to link to external schemas, specifying the URL, version, and directives. The presence of a `@link` directive also indicates that the schema is a core schema.
```
Comment (queerviolet) [inline docs/source/federation-spec.md:338]:
```suggestion
The `@link` directive links definitions within the document to external schemas. External schemas are identified by their `url`, which optionally ends with a name and version in the form `name`/v`major`.`minor`.

The presence of a `@link` directive makes a document a [core schema](https://specs.apollo.dev/#def-core-schema).

The `for` argument describes the purpose of a `@link`. Currently accepted values are `SECURITY` or `EXECUTION`. Core schema-aware servers such as Apollo Router and Gateway will refuse to operate on schemas which contain `@link`s to unsupported specs which are `for: SECURITY` or `for: EXECUTION`.

By default, `@link`ed definitions will be namespaced, i.e. `@federation__requires`. The `import` argument lets you import external definitions into your namespace, so they lose the prefix:
```

File: docs/source/federation-spec.md [hunk 3]
```diff
@@ -309,3 +309,38 @@ extend type User @key(fields: "email") {
...
+  import: [link__Import]
+) repeatable on SCHEMA
+```
+
+The `@link` directive provides a mechanism to link to external schemas, specifying the URL, version, and directives. The presence of a `@link` directive also indicates that the schema is a core schema. 
+
+```graphql
+    schema
+      @link(url: "https://specs.apollo.dev/link/v1.0")
+      @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key", "@requires", "@provides", "@external", { name: "@tag", as "@mytag" }, "@extends", "@shareable", "@inaccessible", "@override"])
```
Comment (queerviolet) [inline docs/source/federation-spec.md:343]:
```suggestion
      @link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@key", "@requires", "@provides", "@external", { name: "@tag", as: "@mytag" }, "@extends", "@shareable", "@inaccessible", "@override"])
```

File: docs/source/federation-spec.md (hunk 1 above)
Comment (queerviolet) [reply docs/source/federation-spec.md:346]: does renaming directives work? we should maybe not talk about it unless it works e2e

Comment (queerviolet) [reply docs/source/federation-spec.md:346]:
```suggestion
In the example above, we import various directives from `federation/v2.0` into our namespace. We also rename one of them, bringing in federation's `@tag` as `@mytag` to distinguish it from a different `@tag` directive already in the schema.
```

**Review by queerviolet — APPROVED** [review]

**Comment by queerviolet** [comment]
looks good, made a couple of suggestions

File: docs/source/federation-spec.md (hunk 2 above)
Comment (benweatherman) [reply docs/source/federation-spec.md:338]: @queerviolet Should that last bit be
> The `as` argument lets you import

I might also word golf that a little
> By default, `@link`ed definitions are namespaced with the name of the schema, e.g. `@federation__requires`. The `as` argument lets you import definitions without the prefix.

**Review by benweatherman — COMMENTED** [review]
🐐 Thanks for getting this started!

File: docs/source/federation-spec.md (hunk 2 above)
Comment (benweatherman) [reply docs/source/federation-spec.md:338]: @queerviolet Should that last bit be
> The `as` argument lets you import

I might also word golf that a little if we're open to it:

> By default, `@link`ed definitions are namespaced with the name of the schema, e.g. `@federation__requires`. The `as` argument lets you import definitions without the prefix.

File: docs/source/federation-spec.md [hunk 4]
```diff
@@ -309,3 +309,38 @@ extend type User @key(fields: "email") {
 ```
 
 This type extension in the Reviews service extends the `User` type from the Users service. It extends it for the purpose of adding a new field called `reviews`, which returns a list of `Review`s.
+
```
[the remark this answers is not in the document]
Comment (clenfest) [reply docs/source/federation-spec.md:312]: Was thinking along the same lines. I'll move it up.

File: docs/source/federation-spec.md [hunk 5]
```diff
@@ -309,3 +309,44 @@ extend type User @key(fields: "email") {
...
+) repeatable on SCHEMA
+```
+
+The `@link` directive links definitions within the document to external schemas. External schemas are identified by their `url`, which optionally ends with a name and version in the form `name`/v`major`.`minor`.
+
+The presence of a `@link` directive makes a document a [core schema](https://specs.apollo.dev/#def-core-schema).
+
+The `for` argument describes the purpose of a `@link`. Currently accepted values are `SECURITY` or `EXECUTION`. Core schema-aware servers such as Apollo Router and Gateway will refuse to operate on schemas which contain `@link`s to unsupported specs which are `for: SECURITY` or `for: EXECUTION`.
+
+By default, `@link`ed definitions will be namespaced, i.e. `@federation__requires`. The `import` argument lets you import external definitions into your namespace, so they lose the prefix:
```
Comment (queerviolet) [inline docs/source/federation-spec.md:344]:
```suggestion
By default, `@link`ed definitions will be namespaced, i.e. `@federation__requires`. The `as` argument lets you pick the name for this namespace:

```graphql
extend schema @link(url: "https://specs.apollo.dev/federation/v2.0". as: "fed2")

type User {
  metrics: Metrics @external
  favorites: UserFavorites @fed2__requires(fields: "metrics")
}
```

The `import` argument lets you import external definitions into your namespace, so they lose the prefix:
```

**Comment by queerviolet** [comment]
also, i don't think it's necessary to dig into the specifics of the scalars and so on here. this isn't a full spec, this is user documentation.

**Review by cpeacock — APPROVED** [review]

**Review by benweatherman — APPROVED** [review]
🐐

File: docs/source/federation-spec.md [hunk 6]
```diff
@@ -252,6 +252,42 @@ type Product @key(fields: "upc") @key(fields: "sku") {
...
+```graphql
+directive @link(
+  url: String, 
+  as: String, 
+  for: link__Purpose, 
+  import: [link__Import]
+) repeatable on SCHEMA
+```
+
+The `@link` directive links definitions within the document to external schemas. External schemas are identified by their `url`, which optionally ends with a name and version in the form `name`/v`major`.`minor`.
```
Comment (rkoron007) [inline docs/source/federation-spec.md:266]: I think the deploy preview made it seem a little unclear what the URL should be, I'd recommend splitting this out into two different sentences, like so:

```
External schemas are identified by their `url`, which optionally ends with a name and version with the following format:

`{NAME}`/v`{MAJOR}`.`{MINOR}`
```

**Review by rkoron007 — APPROVED** [review]
Approved with a few small changes!

File: docs/source/federation-spec.md [hunk 7]
```diff
@@ -252,6 +252,42 @@ type Product @key(fields: "upc") @key(fields: "sku") {
...
+
+The presence of a `@link` directive makes a document a [core schema](https://specs.apollo.dev/#def-core-schema).
+
+The `for` argument describes the purpose of a `@link`. Currently accepted values are `SECURITY` or `EXECUTION`. Core schema-aware servers such as Apollo Router and Gateway will refuse to operate on schemas which contain `@link`s to unsupported specs which are `for: SECURITY` or `for: EXECUTION`.
+
+By default, `@link`ed definitions will be namespaced, i.e. `@federation__requires`. The `as` argument lets you pick the name for this namespace:
+```graphql
+extend schema @link(url: "https://specs.apollo.dev/federation/v2.0". as: "fed2")
```
[the remark this answers is not in the document]
Comment (clenfest) [reply docs/source/federation-spec.md:274]: should be a comma, good catch.

**Pull request merged**